### ISSUE TITLE
New mistral_common package breaks dataset prep

### ISSUE THREAD

**Issue by TheLocalDrummer** [axolotl-ai-cloud/axolotl#3231]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Axolotl throws a warning about having untrainable tokens. Usually handles it properly and gracefully instead of crashing the entire tool.

### Current behaviour

Usually when a dataset has a sample that ends with a message from the user role, Axolotl handles it gracefully by removing the sample or trimming the message out.

Lo and behold, Mistral's new (useless and redundant) tokenizer package `mistral_common` bypasses it and throws this error instead: `Expected last role Assistant for finetuning but got user`

### Steps to reproduce

1. Usually when a dataset has a sample that ends with a message from the user role
2. Axolotl handles it gracefully by removing the sample or trimming the message out.
3. Lo and behold, Mistral's new (useless and redundant) tokenizer package `mistral_common` bypasses it
4. throws this error instead: `Expected last role Assistant for finetuning but got user`

### Config yaml

```yaml

```

### Which Operating Systems are you using?

- [ ] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main-latest as of today

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by TheLocalDrummer** [comment]
Rolling back to weeks ago doesn't fix it.

**Comment by ved1beta** [comment]
Thanks for the bug report, @TheLocalDrummer! I've confirmed this issue on my end.

## Root Cause

This occurs because the `mistral_common` tokenizer package enforces strict validation that conversations must end with an assistant message during fine-tuning.

## workaround

You can work around this by setting the following in your config:

```yaml
tokenizer_use_mistral_common: false

**Comment by NanoCode012** [comment]
Thanks for the report. Could I have more trace of this issue?

We have a check here which should skip the last turn. I didn't expect having mistral-common on or off would change this.

https://github.com/axolotl-ai-cloud/axolotl/blob/bb33fda44d8cc889230698539b8df5a7ba114b67/src/axolotl/prompt_strategies/chat_template.py#L483-L495

Note: If you choose to disable mistral common, it would default to using regular chat_template. Ensure you set the proper template.

**Comment by NanoCode012** [comment]
I just got back to repro this. Error below for future ref. Currently, I would recommend cleaning the dataset beforehand. I think even for non-mistral common cases, we do not recommend last turn to be user (shown in warning above). 

> "Axolotl handles it gracefully by removing the sample or trimming the message out."

To clarify the original post, Axolotl does neither. We just let it pass but print a big warning to users. Ideally, I would prefer erroring though.

I am of the opinion to not auto-drop dataset with bad samples as there can be multiple reasons it occur (ex: slipped through, processing mistake, etc). For prior cases like this, we typically recommend folks to fix their dataset instead.

```
    input_ids = self.prompter.build_prompt(turns, tools=tools)  # type: ignore
  File "/opt/miniconda3/envs/py311/lib/python3.11/site-packages/transformers/tokenization_mistral_common.py", line 1510, in apply_chat_template
    tokenized_request = self.tokenizer.encode_chat_completion(chat_request)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/py311/lib/python3.11/site-packages/mistral_common/tokens/tokenizers/mistral.py", line 364, in encode_chat_completion
    validated_request = self._chat_completion_request_validator.validate_request(request)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/py311/lib/python3.11/site-packages/mistral_common/protocol/instruct/validator.py", line 112, in validate_request
    self.validate_messages(request.messages, continue_final_message=request.continue_final_message)
  File "/opt/miniconda3/envs/py311/lib/python3.11/site-packages/mistral_common/protocol/instruct/validator.py", line 88, in validate_messages
    self._validate_message_list_structure(messages, continue_final_message=continue_final_message)
  File "/opt/miniconda3/envs/py311/lib/python3.11/site-packages/mistral_common/protocol/instruct/validator.py", line 313, in _validate_message_list_structure
    self._validate_last_message(messages[-1], continue_final_message=continue_final_message)
  File "/opt/miniconda3/envs/py311/lib/python3.11/site-packages/mistral_common/protocol/instruct/validator.py", line 385, in _validate_last_message
    super()._validate_last_message(message, continue_final_message)
  File "/opt/miniconda3/envs/py311/lib/python3.11/site-packages/mistral_common/protocol/instruct/validator.py", line 277, in _validate_last_message
    raise InvalidMessageStructureException(
mistral_common.exceptions.InvalidMessageStructureException: Expected last role Assistant for finetuning but got user
